pkgs/development/tools/rust/rust-analyzer/default.nix

```diff
@@ -1,21 +1,36 @@
-{ lib, stdenv, fetchFromGitHub, rustPlatform, CoreServices, cmake
+{ lib, stdenv, fetchFromGitHub, rustPlatform, CoreServices, cmake, fetchpatch
 , libiconv
 , useMimalloc ? false
 , doCheck ? true
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "rust-analyzer-unwrapped";
-  version = "2021-04-19";
-  cargoSha256 = "sha256-CXkI3CQ/v6RBMM2Dpp2u+qnRwba+nqzeaPSJGBiQUoY=";
+  version = "2021-05-10";
+  cargoSha256 = "sha256-PUecBFdYIJFZa5IwwNnuXOkuxtyrzWhxy3C+2jv/hvU=";
 
   src = fetchFromGitHub {
     owner = "rust-analyzer";
     repo = "rust-analyzer";
     rev = version;
-    sha256 = "sha256-W/cUwZEvlUXzqQ/futeNFwDWR/cTL/RLZaW2srIs83Q=";
+    sha256 = "sha256-oz6FqRMEUUTS4X2XhpWjp2JIgl1A6wQv2OU8auwUoVM=";
   };
 
+  patchFlags = [ "-Rp1" ];
```

Comment:
Please use revert = true instead of patchFlags https://github.com/NixOS/nixpkgs/blob/d2ea330f62442af11df60e51f00299bf735126b5/pkgs/desktops/gnome/core/gnome-shell/default.nix#L104